awx/main/migrations/0077_v360_replace_job_origin.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.20 on 2019-05-14 14:51
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('main', '0076_v360_add_new_instance_group_relations'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='joborigin',
+            name='instance',
+        ),
+        migrations.RemoveField(
+            model_name='joborigin',
+            name='unified_job',
+        ),
+        migrations.AddField(
+            model_name='activitystream',
+            name='action_node',
+            field=models.CharField(blank=True, default='', editable=False, help_text='The cluster node the activity took place on.', max_length=512),
+        ),
+        migrations.DeleteModel(
```

Comment:
Do we have any idea how long this might take for a `JobOrigin` table w/ 100M rows?  I'll bet certain users with lots of historical job launches have some truly gigantic tables here.

Reply:
My hope is that it comes out like

```sql
DROP TABLE main_joborigin CASCADE
```

from looking at

https://github.com/django/django/blob/b69f8eb04cc8762d3dfd5af5ea1fc58e3f2ebcc3/django/db/backends/base/schema.py#L324-L327

I don't have any reason to think that it does any looping over them, but I also don't have a good approach to profiling the operations to verify this.

Reply:
What gets printed if you run:

```
awe-manage sqlmigrate main 0079_v360_replace_job_origin
```

Reply:
Firstly I need to confirm that I have more than 0 JobOrigin entries in my system currently, since running this and reversing it will leave them missing.

```python
In [1]: JobOrigin.objects.count()
Out[1]: 17
```

This established, I can run your command.

```sql
BEGIN;
--
-- Remove field instance from joborigin
--
SET CONSTRAINTS "main_joborigin_instance_id_ac57e660_fk_main_instance_id" IMMEDIATE; ALTER TABLE "main_joborigin" DROP CONSTRAINT "main_joborigin_instance_id_ac57e660_fk_main_instance_id";
ALTER TABLE "main_joborigin" DROP COLUMN "instance_id" CASCADE;
--
-- Remove field unified_job from joborigin
--
SET CONSTRAINTS "main_joborigin_unified_job_id_52f721cb_fk_main_unifiedjob_id" IMMEDIATE; ALTER TABLE "main_joborigin" DROP CONSTRAINT "main_joborigin_unified_job_id_52f721cb_fk_main_unifiedjob_id";
ALTER TABLE "main_joborigin" DROP COLUMN "unified_job_id" CASCADE;
--
-- Add field action_node to activitystream
--
ALTER TABLE "main_activitystream" ADD COLUMN "action_node" varchar(512) DEFAULT '' NOT NULL;
ALTER TABLE "main_activitystream" ALTER COLUMN "action_node" DROP DEFAULT;
--
-- Delete model JobOrigin
--
DROP TABLE "main_joborigin" CASCADE;
COMMIT;
```

At a quick look, this appears to be the same result if I go back and forward again 0079->0078->0079. The last 0078->0079 is done, of course, with no JobOrigin records.